### PR TITLE
[Model Monitoring] Add `rebuild_images` param to `enable_model_monitoring()`

### PR DESCRIPTION
**Pull request by davesh0812**
https://iguazio.atlassian.net/browse/ML-6580

### CODE REVIEW

File: tests/system/model_monitoring/test_app.py [hunk 1]
```diff
@@ -725,6 +725,43 @@ def test_model_monitoring_crud(self) -> None:
...
+            func = self.project.get_function(
+                key=name,
+                ignore_cache=True,
+            )
+            func._get_db().get_nuclio_deploy_status(func, verbose=False)
+            assert func.status.state == "building"
+
+        self.project._wait_for_functions_deployment(
+            mm_constants.MonitoringFunctionNames.list()
+            + [mm_constants.HistogramDataDriftApplicationConstants.NAME]
```
Comment (jond01) [inline tests/system/model_monitoring/test_app.py]: Consider using some variable for these functions, it repeats here three times

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -2116,6 +2116,7 @@ def enable_model_monitoring(
         *,
         deploy_histogram_data_drift_app: bool = True,
         wait_for_deployment: bool = False,
+        overwrite: bool = False,
```
Comment (jond01) [inline mlrun/projects/project.py]: Please add the overwrite argument also to the `deploy_histogram_data_drift_app` function.

**Review by jond01 — APPROVED** [review]
Please address the last comment, other than that it looks good

File: mlrun/projects/project.py (hunk 2 above)
Comment (davesh0812) [reply mlrun/projects/project.py]: I think ׳e can add it in different pr, not sure we want to add this option to the user.

File: mlrun/db/httpdb.py [hunk 3]
```diff
@@ -3329,13 +3330,14 @@ def enable_model_monitoring(
...
-                            stream functions, which are real time nuclio functions.
-                            By default, the image is mlrun/mlrun.
+        :param project:                         Project name.
+        :param base_period:                     The time period in minutes in which the model monitoring controller
+                                                function triggers. By default, the base period is 10 minutes.
+        :param image:                           The image of the model monitoring controller, writer & monitoring
+                                                stream functions, which are real time nuclio functions.
+                                                By default, the image is mlrun/mlrun.
         :param deploy_histogram_data_drift_app: If true, deploy the default histogram-based data drift application.
+        :param rebuild_images:                  If True, it would overwrite the existing model monitoring functions.
```
Comment (assaf758) [inline mlrun/db/httpdb.py]:
```suggestion
        :param rebuild_images:                  If true, force rebuild of model monitoring infrastructure images.
```

File: mlrun/projects/project.py [hunk 4]
```diff
@@ -2135,6 +2136,7 @@ def enable_model_monitoring(
         :param wait_for_deployment:             If true, return only after the deployment is done on the backend.
                                                 Otherwise, deploy the model monitoring infrastructure on the
                                                 background, including the histogram data drift app if selected.
+        :param rebuild_images:                  If true, overwrite the existing model monitoring infrastructure images.
```
Comment (assaf758) [inline mlrun/projects/project.py]:
```suggestion
        :param rebuild_images:                  If true, force rebuild of model monitoring infrastructure images.
```

File: server/api/api/endpoints/model_monitoring.py [hunk 5]
```diff
@@ -100,13 +101,15 @@ async def enable_model_monitoring(
...
-                        By default, the image is mlrun/mlrun.
+    :param commons:                         The common parameters of the request.
+    :param base_period:                     The time period in minutes in which the model monitoring controller function
+                                            triggers. By default, the base period is 10 minutes.
+    :param image:                           The image of the model monitoring controller, writer & monitoring
+                                            stream functions, which are real time nuclio functions.
+                                            By default, the image is mlrun/mlrun.
     :param deploy_histogram_data_drift_app: If true, deploy the default histogram-based data drift application.
+    :param rebuild_images:                  If True, it would overwrite the existing model monitoring controller,
+                                            writer & stream functions.
```
Comment (assaf758) [inline server/api/api/endpoints/model_monitoring.py]:
```suggestion
    :param rebuild_images:                  If true, force rebuild of model monitoring infrastructure images (controller, writer & stream).
```

File: server/api/crud/model_monitoring/deployment.py [hunk 6]
```diff
@@ -550,41 +556,50 @@ def _check_if_already_deployed(
                 pass
         return False
 
-    def deploy_histogram_data_drift_app(self, image: str) -> None:
+    def deploy_histogram_data_drift_app(
+        self, image: str, overwrite: bool = False
```
Comment (assaf758) [inline server/api/crud/model_monitoring/deployment.py:561]: overwrite => rebuild_image

**Review by assaf758 — CHANGES_REQUESTED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**